src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/hotshard/HotShardClusterRca.java

```diff
@@ -90,17 +95,21 @@ private void populateResourceInfoTable(String indexName, NodeShardKey nodeShardK
     private void consumeFlowUnit(ResourceFlowUnit resourceFlowUnit) {
         String nodeId = ((HotNodeSummary) resourceFlowUnit.getResourceSummary()).getNodeID();
         HotNodeSummary hotNodeSummary = ((HotNodeSummary) resourceFlowUnit.getResourceSummary());
-        for (HotShardSummary hotShardSummary : hotNodeSummary.getHotShardSummaryList()) {
-            String indexName = hotShardSummary.getIndexName();
-            NodeShardKey nodeShardKey = new NodeShardKey(nodeId, hotShardSummary.getShardId());
-            // 1. Populate CPU Table
-            populateResourceInfoTable(indexName, nodeShardKey, hotShardSummary.getCpuUtilization(), cpuUtilizationInfoTable);
+        for (GenericSummary summary : hotNodeSummary.getNestedSummaryList()) {
+            if (summary instanceof HotShardSummary) {
```

Comment:
We have a lot of `instanceof` scattered around. It might be worth while to create an `enum` type of Summaries and then we can compare against an enum value

Reply:
Agreed, have created an issue to track it https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/issues/212